go/ephemeral/user_ek_box_storage.go

```diff
@@ -179,21 +179,21 @@ func (s *UserEKBoxStorage) Put(ctx context.Context, generation keybase1.EkGenera
 	return nil
 }
 
-func (s *UserEKBoxStorage) unbox(ctx context.Context, userEKBoxed keybase1.UserEkBoxed) (userEK keybase1.UserEk, err error) {
+func (s *UserEKBoxStorage) unbox(ctx context.Context, userEKGeneration keybase1.EkGeneration, userEKBoxed keybase1.UserEkBoxed) (userEK keybase1.UserEk, err error) {
 	defer s.G().CTrace(ctx, "UserEKBoxStorage#unbox", func() error { return err })()
 
 	deviceEKStorage := s.G().GetDeviceEKStorage()
 	deviceEK, err := deviceEKStorage.Get(ctx, userEKBoxed.DeviceEkGeneration)
 	if err != nil {
-		return userEK, err
+		return userEK, newEKUnboxErr(USER_EK, userEKGeneration, DEVICE_EK, userEKBoxed.DeviceEkGeneration)
 	}
 
 	deviceSeed := DeviceEKSeed(deviceEK.Seed)
 	deviceKeypair := deviceSeed.DeriveDHKey()
 
 	msg, _, err := deviceKeypair.DecryptFromString(userEKBoxed.Box)
 	if err != nil {
-		return userEK, err
+		return userEK, newEKUnboxErr(USER_EK, userEKGeneration, DEVICE_EK, userEKBoxed.DeviceEkGeneration)
```

Comment:
I wonder if these errors should get the same special treatment as the team ones, since the team ones are expected (come out of the closet and read a chat message) but these aren't (you'd need a box that was encrypted for a device key you never made / deleted early). Then again, maybe we can express that difference when we're catching them? Let me know what you think.

Reply:
There are two cases handled by these errors, for one I think users/teams should be handled the same since you ultimately need a new EK. If the `Get` calls to (deviceEK, userEBox) storage fails, could have a User/teamEK that your device/user doesn't have access to (maybe you were stale when it was created) so you need to make new keys.

In the case that decryption fails (either for user or teams) it would be much stranger (since you can retrieve/decrypt the secret from storage, but the box itself is corrupted and you would still need a new EK.

 I added some warning logs wherever we return these errors -- should we handle in any other way?